src/test/java/seedu/address/logic/parser/EditCommandParserTest.java

```diff
@@ -1,211 +1,480 @@
+
+        // missing all fields in note prefix
+        assertParseFailure(parser, VALID_NOTE_INDEX + ITEM_TYPE_NOTE,
+                expectedNoteErrorMessage);
+
+        String expectedProjectErrorMessage =
+                EditProjectCommand.MESSAGE_NOT_EDITED;
+
+        // missing all fields in project prefix
+        assertParseFailure(parser, VALID_PROJECT_INDEX + ITEM_TYPE_PROJECT,
+                expectedProjectErrorMessage);
+
+        String expectedResumeErrorMessage =
+                EditResumeCommand.MESSAGE_NOT_EDITED;
+
+        // missing name prefix
+//        assertParseFailure(parser, ITEM_TYPE_RESUME + PREFIXED_TAG_JAVA, expectedResumeErrorMessage);
+
+        String expectedSkillErrorMessage =
+                EditSkillCommand.MESSAGE_NOT_EDITED;
+
+        // missing all fields in skill prefix
+        assertParseFailure(parser, VALID_SKILL_INDEX + ITEM_TYPE_SKILL,
+                expectedSkillErrorMessage);
+    }
+
+    @Test
+    public void parse_invalidValue_failure() {
+        // invalid name
+        assertParseFailure(parser, VALID_INTERNSHIP_INDEX +
+                ITEM_TYPE_INTERNSHIP + INVALID_NAME_DESC + PREFIXED_ROLE_FRONTEND + PREFIXED_TIME_FROM
+                        + PREFIXED_TIME_TO + PREFIXED_INTERNSHIP_DESCRIPTION + PREFIXED_TAG_FRONTEND,
+                Name.MESSAGE_CONSTRAINTS);
+
+        // invalid from
+        assertParseFailure(parser, VALID_INTERNSHIP_INDEX +
+                ITEM_TYPE_INTERNSHIP + PREFIXED_NAME_GOOGLE + PREFIXED_ROLE_FRONTEND + INVALID_FROM_DESC
+                        + PREFIXED_TIME_TO + PREFIXED_INTERNSHIP_DESCRIPTION + PREFIXED_TAG_FRONTEND,
+                Time.MESSAGE_CONSTRAINTS);
+
+        // invalid TO
+        assertParseFailure(parser, VALID_INTERNSHIP_INDEX +
+                ITEM_TYPE_INTERNSHIP + PREFIXED_NAME_GOOGLE + PREFIXED_ROLE_FRONTEND + PREFIXED_TIME_FROM
+                        + INVALID_TO_DESC + PREFIXED_INTERNSHIP_DESCRIPTION + PREFIXED_TAG_FRONTEND,
+                Time.MESSAGE_CONSTRAINTS);
+
+        // invalid tag
+        assertParseFailure(parser, VALID_INTERNSHIP_INDEX +
+                ITEM_TYPE_INTERNSHIP + PREFIXED_NAME_GOOGLE + PREFIXED_ROLE_FRONTEND + PREFIXED_TIME_FROM
+                        + PREFIXED_TIME_TO + PREFIXED_INTERNSHIP_DESCRIPTION + INVALID_TAG_DESC,
+                Tag.MESSAGE_CONSTRAINTS);
+
+        // invalid website
+        assertParseFailure(parser, VALID_INTERNSHIP_INDEX +
+                ITEM_TYPE_PROJECT + PREFIXED_NAME_ORBITAL + PREFIXED_TIME_TO
+                        + INVALID_WEBSITE_DESC + PREFIXED_DESCRIPTION_ORBITAL,
+                Website.MESSAGE_CONSTRAINTS);
+
+        // invalid level
+        assertParseFailure(parser, VALID_INTERNSHIP_INDEX +
+                ITEM_TYPE_SKILL + PREFIXED_NAME_REACT + INVALID_LEVEL_DESC + PREFIXED_TAG_FRONTEND,
+                "Level of proficiency can only be one of these three types: basic, intermediate, "
```

Comment:
This is to be updated now that the `Level` class has its own `MESSAGE_CONSTRAINTS` field.